.github/workflows/build-notebooks-TEMPLATE.yaml

```diff
@@ -235,6 +235,121 @@ jobs:
+
+          sudo systemctl start crio.service
+        env:
+          CRIO_VERSION: v1.30
+          KUBERNETES_VERSION: v1.30
+
+      - name: Show crio debug data (on failure)
+        if: ${{ failure() && steps.have-tests.outputs.tests == 'true' }}
+        run: |
+          set -Eeuxo pipefail
+
+          sudo systemctl status crio.service || true
+          sudo journalctl -xeu crio.service
+
+      # do this early, it's a good check that cri-o is not completely broken
+      - name: "Show crio images information"
+        if: ${{ steps.have-tests.outputs.tests == 'true'  }}
+        run: sudo crictl images
+
+      - name: Install Kubernetes cluster
+        if: ${{ steps.have-tests.outputs.tests == 'true'  }}
+        run: |
+          set -Eeuxo pipefail
+
+          sudo swapoff -a
+          sudo modprobe br_netfilter
+          sudo sysctl -w net.ipv4.ip_forward=1
+
+          # Was getting strange DNS resolution errors from pods that don't seem to want to go away sometimes:
+          #  Resolving raw.githubusercontent.com (raw.githubusercontent.com)... failed: Name or service not known.
+          #  wget: unable to resolve host address ‘raw.githubusercontent.com’
+          # Here's what helped:
+          #  https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/#known-issues
+          #  https://github.com/kubernetes/kubernetes/blob/e4c1f980b76fecece30c2f77885a7117192170a6/CHANGELOG/CHANGELOG-1.30.md?plain=1#L1454
+          #  https://github.com/canonical/microk8s/issues/68#issuecomment-404923563
+          sudo ufw allow in on cni0
+          sudo ufw allow out on cni0
+          sudo ufw default allow routed
+          sudo iptables -P FORWARD ACCEPT
+          sudo iptables -t nat -A POSTROUTING -s 10.85.0.0/16 -o eth0 -j MASQUERADE
+
+          # https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm
+          sudo kubeadm init --config=ci/cached-builds/kubeadm.yaml
+
+          mkdir -p $HOME/.kube
+          sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+          sudo chown $(id -u):$(id -g) $HOME/.kube/config
+
+      - name: Show kubelet debug data (on failure)
+        if: ${{ failure() && steps.have-tests.outputs.tests == 'true' }}
+        run: |
+          set -Eeuxo pipefail
+
+          sudo systemctl status kubelet || true
+          sudo journalctl -xeu kubelet
+
+          # Here is one example how you may list all running Kubernetes containers by using crictl:
+          sudo crictl --runtime-endpoint unix:///var/run/crio/crio.sock ps -a | grep kube | grep -v pause
+          # Once you have found the failing container, you can inspect its logs with:
+          # crictl --runtime-endpoint unix:///var/run/crio/crio.sock logs CONTAINERID
+
+      - name: Untaint the master
+        if: ${{ steps.have-tests.outputs.tests == 'true' }}
+        run: kubectl taint nodes --all node-role.kubernetes.io/control-plane-
```

Comment:
Just checking that this should end with the `-` sign indeed?

Reply:
We should, that's just how you remove taints in kuberbetes. Somewhat similar to deleting branches in git with the branch names prefaced with :

Reply:
Ah, I see; pardon my ignorance than, thanks!

Reply:
No worries, we do these reviews to increase familiarity, right? Catching mistakes is only secondary

The k8s docs is very particular, telling people not to miss the - when copypasting from it ;)